### PR TITLE
EVG-11870: Content Recommendations Card global template

### PR DESCRIPTION
**Pull request by nriserEG**
## Global Template: Content Recommendations Card

JIRA: https://help.evergage.com/browse/EVG-11870
This global template inserts recommended contents in a two-column format, based on a selected Einstein Recipe.

**TODOs / WIP:**
- ~~Update content zones in Serverside code to standardized names (waiting to hear back from Kevin)~~ (The missing content zones will be added for V2)
- ~~Add png file of screen shot of the template rendered on cumulus page~~
- ~~Add template info and screen shots to README~~
- ~~Update/finalize comments in Handlebars~~
- ~~Ask Kyle for a review (business user perspective)~~ Got his feedback and made updates

**Note:**
- Cumulus does not have proper blog post pages, so I created dummy blog posts in the Catalog. @kylerobbins is trying to work with the team/person who is in charge of updating cumulus to include blog post pages 
- I built this template to be generally consistent with the Product Recommendations Card global template that Andy is currently working on ([PR](https://github.com/evergage/evergage-global-templates/pull/3)). I've implemented the same feedback (both my own and others) from the PR into this Content Recs template.

### Things to check for:
* **Visuals:** Does it look good across different client sites? To test this, you can use this snippet https://gist.github.com/nriserEG/e632356d5b9bc61ad92e5e3b07dda3ba in the Chrome inspector tool that injects the minified template HTML and CSS code (the template will appear at the very top of the page you test on).
* **Deferred rendering:** Is the rendering of the template deferred until the content zone element is loaded on page? 
* **Flicker defense:** Is the template flicker-defensed? Promise must be returned in the apply function for flicker defense to work. 
* **Campaign stats tracking:** Do impression stats and click stats (upon clicking the CTA) fire for both the control and test experience?
* **Template-specific documentation**: Is there sufficient information for a template developer to use the global template and customize for their own use?
* **WCAG compliant:** Does the template contain ARIA attributes and use semantic elements for web accessibility? 
** FYI for this template specifically, I included aria-label attributes and adds fallback background colors in case the background image fails to load.
** You can use the WAVE Chrome extension tool https://chrome.google.com/webstore/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh for a quick check. 
* **Mobile/smaller screen responsiveness:** Does the campaign generally look good on smaller screens?

**Important note:** In order to test **Flicker Defense** properly, we want to test the template on a site that already loads Beacon v16. The cumulus demo site does not have the beacon loaded yet and it’s not clear when it’ll be added, so please test on the Dev Docs site.

#### Template: 
- You can view the demo template in eng1:globaltemplates_cumulus, **template name: [EVG-11870] Global - Recipe Content Recommendations**. 

#### Campaign: 
- You can view the demo campaign using this template in eng1:globaltemplates, **campaign name: [EVG-11870] Global - Recipe Content Recommendations** (campaign is set to show up on the Homepage, recs row)
- Sample cumulus test link: https://cumulusfinserv.com/?evergageTestMessages=h0rrL

Note: If you test on cumulus, you will have to inject the beacon.

### CODE REVIEW

File: recipe-content-recommendations/template.hbs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+aria-label="Content Recommendations {{#if header}}- {{header}}{{/if}}">
+    <h3 class="evg-header evg-h3">{{header}}</h3>
+    <div class="evg-content-recommendations evg-container">
+        {{#each blogs}}
+        <div class="evg-card-container">
+            <article class="evg-content-recommendation evg-card" data-evg-item-id="{{id}}" 
+            data-evg-item-type="{{../itemType}}">
+                <a href="{{attributes.url.value}}">
+                    <img class="evg-content-image evg-card-image" 
+                    data-evg-product-imageUrl="{{attributes.imageUrl.value}}" src="{{attributes.imageUrl.value}}"
```
Comment (dcohenEG) [inline recipe-content-recommendations/template.hbs:31]: what is the purpose of `data-evg-product-imageUrl`?

Comment (nriserEG) [reply recipe-content-recommendations/template.hbs:31]: My bad, old code. Thanks for the catch!

File: recipe-content-recommendations/template.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    @header('Recommendation Attributes')
+
+    @title("Show content name")
+    nameVisibility: boolean = true;
+
+    @title("Show content description")
+    descriptionVisibility: boolean = true;
+
+    @hidden(true)
+    visibilityOptions: object;
```
[the remark this answers is not in the document]
Comment (nriserEG) [reply recipe-content-recommendations/template.ts:39]: Ah this was part of an old implementation. Thanks for the catch!

File: recipe-content-recommendations/template.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+     */
+    @options([
+        "home_recs",
+        "search_recs"
+    ])
+    contentZone: string = "home_recs";
+
+    @title("Recommendations Block Title")
+    @subtitle("i.e. You May Also Like")
+    header: string = "You May Also Like";
```
[the remark this answers is not in the document]
Comment (nriserEG) [reply recipe-content-recommendations/template.ts:22]: Thanks for the suggestion, but Andy and I decided to go with a more general name for now for both recs templates.

**Pull request closed without merging**